src/rendering/vcFenceRenderer.cpp

```diff
@@ -60,44 +60,60 @@ static vcTexture *gSolidTexture = nullptr;
 static vcTexture *gDiagonalTexture = nullptr;
 
 udResult vcFenceRenderer_CreateSegmentVertexData(vcFenceRenderer *pFenceRenderer, vcFenceSegment *pSegment);
+udResult vcFenceRenderer_Init();
+udResult vcFenceRenderer_Destroy();
 
-void vcFenceRenderer_Init()
+udResult vcFenceRenderer_Init()
 {
+  udResult result;
   gRefCount++;
-  if (gRefCount == 1)
-  {
-    vcTexture_CreateFromFilename(&gArrowTexture, "asset://assets/textures/fenceArrow.png", nullptr, nullptr, vcTFM_Linear, true);
-    vcTexture_CreateFromFilename(&gGlowTexture, "asset://assets/textures/fenceGlow.png", nullptr, nullptr, vcTFM_Linear, true);
-    vcTexture_CreateFromFilename(&gDiagonalTexture, "asset://assets/textures/fenceDiagonal.png", nullptr, nullptr, vcTFM_Linear, true);
+  static const uint32_t whitePixel[] = { 0xffffffff };
 
-    static const uint32_t solidPixels[] = { 0xffffffff };
-    vcTexture_Create(&gSolidTexture, 1, 1, solidPixels);
-  }
+  UD_ERROR_IF(gRefCount != 1, udR_Success);
+
+  UD_ERROR_IF(!vcTexture_CreateFromFilename(&gArrowTexture, "asset://assets/textures/fenceArrow.png", nullptr, nullptr, vcTFM_Linear, true), udR_InternalError);
+  UD_ERROR_IF(!vcTexture_CreateFromFilename(&gGlowTexture, "asset://assets/textures/fenceGlow.png", nullptr, nullptr, vcTFM_Linear, true), udR_InternalError);
+  UD_ERROR_IF(!vcTexture_CreateFromFilename(&gDiagonalTexture, "asset://assets/textures/fenceDiagonal.png", nullptr, nullptr, vcTFM_Linear, true), udR_InternalError);
+
+  UD_ERROR_CHECK(vcTexture_Create(&gSolidTexture, 1, 1, whitePixel));
+
+  result = udR_Success;
+epilogue:
+
+  if (result != udR_Success)
+    vcFenceRenderer_Destroy();
+  return result;
 }
 
-void vcFenceRenderer_Destroy()
+udResult vcFenceRenderer_Destroy()
 {
+  udResult result;
   --gRefCount;
-  if (gRefCount == 0)
-  {
-    vcTexture_Destroy(&gArrowTexture);
-    vcTexture_Destroy(&gGlowTexture);
-    vcTexture_Destroy(&gSolidTexture);
-    vcTexture_Destroy(&gDiagonalTexture);
-  }
+
+  UD_ERROR_IF(gRefCount != 0, udR_Success);
+
+  vcTexture_Destroy(&gArrowTexture);
+  vcTexture_Destroy(&gGlowTexture);
+  vcTexture_Destroy(&gSolidTexture);
+  vcTexture_Destroy(&gDiagonalTexture);
+
+  result = udR_Success;
+epilogue:
+
+  return result;
```

Comment:
This only returns success?

Reply:
Yes. Intentional so future additions to the function won't need error handling code added back in.